src/commands/dev.ts

```diff
@@ -100,19 +103,49 @@ const sendFile = (
   ext = '.html',
 ) => {
   const ETag = etag(body);
-  const headers = {
+  const headers: Record<string, string> = {
     'Content-Type': mime.contentType(ext) || 'application/octet-stream',
     'Access-Control-Allow-Origin': '*',
     ETag,
+    Vary: 'Accept-Encoding'
   };
 
   if (req.headers['if-none-match'] === ETag) {
     res.writeHead(304, headers);
+    res.end()
+    return
+  }
+
+  let acceptEncoding = (req.headers['accept-encoding'] as string) || ''
+  if (req.headers["cache-control"]?.includes('no-transform') || ['HEAD', 'OPTIONS'].includes(req.method!) || !isCompressible(mime.contentType(ext))) {
+    acceptEncoding = ''
+  }
+
+  function onError(err) {
+    if (err) {
+      res.end()
+      console.error('An error occurred:', err)
+    }
+  }
+
+  const raw = stream.Readable.from([body])
+  if (/\bgzip\b/.test(acceptEncoding)) {
+    headers['Content-Encoding'] = 'gzip'
+    res.writeHead(200, headers)
+    stream.pipeline(raw, zlib.createGzip(), res, onError)
+  } else if (/\bbr\b/.test(acceptEncoding)) {
```

Comment:
Yea, lets remove brotli & deflate for now. It's great that it was easy to add but my concern is:

- Brotli is slower / more expensive to run than gzip. My understanding is that its great for staticly hosted files but less when you need to run that compression every time you serve it.
- Brotli isn't supported in all versions of Node v10.

My goal with this feature is just to give users better insight of how their site performs in production. The difference between no compression and gzip is valuable towards this goal, but the difference between gzip and brotli is much smaller.